### PR TITLE
Upgrade to OpenGSN v2.1

### LINKED ISSUE #81: Implement custom GSN Forwarder to avoid double signature checks
Right, we're double checking signatures by using the default Forwarder. We can implement our own dummy forwarder which ignores the signature on the call, because we know it will be checked inside the `withdrawTokenOnBehalf` method.

See also #20 to avoid relaying calls to methods other than `withdrawTokenOnBehalf`. Open question: should this happen in the Paymaster or the Forwarder?

### LINKED ISSUE #86: Upgrade to Production Version of OpenGSN Packages
See branch https://github.com/ScopeLift/umbra-protocol/tree/gsn-21-upgrade

After upgrading, GSN tests fail with:

```
-Failed to relay call. Results:
-Relaying errors (1):
 -http://127.0.0.1:64085 => paymaster rejected in local view call to 'relayCall()' : signature mismatch
```

### PR DESCRIPTION
**Pull request by apbendi**
This PR upgrades OpenGSN to the latest mainnet version, along with the tests for OpenGSN support. It also cleans up the `UmbraForwarder` stub to include the absolute minimum set of functionality needed to comply with the GSN architecture without actually doing the validation inside the forwarder contract.

closes #86 
closes #81

### CODE REVIEW

File: contracts/contracts/open_gsn/UmbraPaymaster.sol [hunk 1]
```diff
@@ -14,7 +14,7 @@ contract UmbraPaymaster is BasePaymaster {
   }
 
   function versionPaymaster() external view override returns (string memory) {
-    return "1.0.0";
+    return "2.0.0";
```
Comment (mds1) [inline contracts/contracts/open_gsn/UmbraPaymaster.sol:17]: Do we actually need to bump this to 2.0.0 given that it's never been used for anything real? It looks like this method is not used in signatures anywhere so it shouldn't affect replay attacks

File: contracts/contracts/open_gsn/UmbraPaymaster.sol [hunk 2]
```diff
@@ -26,7 +26,7 @@ contract UmbraPaymaster is BasePaymaster {
     (signature, approvalData, maxPossibleGas); // to silence compiler warnings
 
     require(relayRequest.request.to == umbraAddr, "UmbraPaymaster: Not Target");
-    return (abi.encode(0x0), true);
+    return ("", true);
```
Comment (mds1) [inline contracts/contracts/open_gsn/UmbraPaymaster.sol:29]: What's the rationale behind this change?

File: contracts/test/umbra-gsn-test.js [hunk 3]
```diff
@@ -39,24 +38,29 @@ describe('Umbra GSN', () => {
...
     ctx.forwarder = deployedForwarder.address;
 
-    // Deploy the Umbra GSN wrapper & paymaster
+    // Deploy the Umbra GSN wrapper
     ctx.relayRecipient = await UmbraRelayRecipient.new(ctx.umbra.address, ctx.forwarder, {
       from: owner,
     });
+
+    // Deploy the Umbra Paymaster
     ctx.paymaster = await UmbraPaymaster.new(ctx.relayRecipient.address, { from: owner });
```
Comment (mds1) [inline contracts/test/umbra-gsn-test.js:57]: Didn't we migrate to hardhat? It looks like we still use a lot of web3 in the tests. Maybe we didn't migrate to ethers-style and just changed the VM?

**Review by mds1 — APPROVED** [review]
Test pass, looks good! Just some small comments

File: contracts/test/umbra-gsn-test.js (hunk 3 above)
Comment (wildmolasses) [reply contracts/test/umbra-gsn-test.js:57]: Yes, we are using hardhat, and the hardhat-web3 plugin. Tests could be updated to use ethers, but it wasn't strictly necessary!

File: contracts/contracts/open_gsn/UmbraPaymaster.sol (hunk 1 above)
Comment (apbendi) [reply contracts/contracts/open_gsn/UmbraPaymaster.sol:17]: Good question. I'm not exactly sure what was causing it, but something was throwing an error until I updated this value. I think perhaps their JS library (used in the tests) for interacting with the contracts will throw if the versions don't agree.

File: contracts/contracts/open_gsn/UmbraPaymaster.sol (hunk 2 above)
Comment (apbendi) [reply contracts/contracts/open_gsn/UmbraPaymaster.sol:29]: I just pulled this from their updated Paymaster reference implementation for the mainnet v2.0. Not sure it actually matters.
